src/component/FormFields/Attribute.js

```diff
@@ -0,0 +1,104 @@
+import React, {
+   Component
+ } from 'react';
+import Select from 'react-select';
+import 'react-select/dist/react-select.css';
+import data from '@groceristar/groceristar-fetch/search';
+
+let attribute;
+
+function toOpt(arr) {
+    let Opt= arr.reduce((intermediate, item, index) => {
+        intermediate[index]={};
+        intermediate[index].value = index;
+        intermediate[index].label = item;
+        return intermediate;
+    }, []);
+    return Opt;
+}
+
+
+
+
+class Attribute extends Component {
+  constructor(props) {
+        super(props);
+        this.state = {
+            value: undefined,
+            options: []
+        };
+
+        this.handleOnChange = this.handleOnChange.bind(this);
+        this.getAttributeData = this.getAttributeData.bind(this);
+        this.getPlaceholder = this.getPlaceholder.bind(this);
+    }
+
+    handleOnChange(value) {
+        this.setState({
+            value
+        });
+    }
+    
+    
+    
+    getAttributeData() {
```

Comment:
This function is not what i'm proud of. i'm pretty sure it can be done in a better way. Same thing for a placeholder function

Reply:
@atherdon Okay I too feel that it is not the best function, but the question is how? Would using 'refs' help? I mean at the end of the day we have to call one of the data.get functions anyway, we can use switch case but that is deprecated I guess! Or we can do this check on the props under the hood, but I doubt it will make any difference.

Reply:
i'm not good with `refs` so i cannot advice.
it can be switch or array/object